hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -1558,6 +1575,18 @@ def draw_bytes(self, size: int, *, forced: Optional[bytes] = None) -> bytes:
         return self.provider.draw_bytes(size, forced=forced)
 
     def draw_boolean(self, p: float = 0.5, *, forced: Optional[bool] = None) -> bool:
+        # Internally, we treat probabilities lower than 1 / 2**64 as
+        # unconditionally false. These conditionals should be checking against
+        # 64 bits, but we need some extra bits of safety in practice, likely due
+        # to internal floating point errors during computation.
+        #
+        # Note that even if we lift this 64 bit restriction in the future, p
+        # cannot be 0 (1) when forced is True (False).
+        if forced is True:
+            assert p > 2 ** (-62)
+        if forced is False:
+            assert p < (1 - 2 ** (-62))
```

Comment:
Ah, floating-point imprecision strikes again... I think the problem with using different bounds here is that it means we can draw something which is then impossible to force!  Of course that's very (very very very...) unlikely to happen by chance, but we intend to have a SMT-backed provider and replaying a failing example to shrink could easily hit this.

What happens if we use 2**64 here?  Can we fix those errors instead?

Reply:
apparently this works fine at 64 bits? 😅 not sure where I messed up here. The tests clearly pass when I lift this restriction though, so I've done so.